state/secrets.go

```diff
@@ -634,7 +634,7 @@ func (st *State) deleteSecrets(uris []*secrets.URI, revisions ...int) (external
 			}
 			revisionRegexp := fmt.Sprintf("(%s)", strings.Join(revs, "|"))
 			_, err = secretRevisionsCollection.Writeable().RemoveAll(bson.D{{
-				"_id", bson.D{{"$regex", fmt.Sprintf("%s/%s", uri.ID, revisionRegexp)}},
+				"_id", bson.D{{"$regex", fmt.Sprintf("%s/%s$", uri.ID, revisionRegexp)}},
```

Comment:
Should we also include the optimisation from here https://github.com/juju/juju/issues/20746